### ISSUE TITLE
suggestion: adding a more detailed data pipeline instruction on fine-tuning colab tutorial 

### ISSUE THREAD

**Issue by ziz19** [speechbrain/speechbrain#694]
It's been great to see more tutorials being added to learn speechbrain. For this old tutorial [Pretrained Models and Fine-Tuning with huggface](https://colab.research.google.com/drive/1LN7R3U3xneDgDRK2gC5MzGkLysCWxuC3?usp=sharing]), I notice there is slight inconsistency between the loading data procedure to use pre-trained model and to fine-tune the model. 

In using pre-trained model to transcribe the file, the audio is loaded using the `EncoderDecoderASR` interface, which will 1) load the audio using torchaudio 2) normalize the audio using an audio normalizer. 

In the fine-tuning section, the audio is loaded using `dataset.add_dynamic_item(sb.dataio.dataio.read_audio, takes="file_path", provides="signal")`, and no normalizer is used within the function. 

Since the decoding relies on normalized audio, wouldn't it be more natural to fine-tune the model based on normalized audio rather than raw audio?

 It might also be useful to add a section about how to use the fine-tuned model for transcribing file.

**Comment by ziz19** [comment]
Another update:
The fine-tuning procedure is not updating my weights. The loss of each epoch remains the same. After some exploration, I found that in pretrained model, all layer weights are frozen(`requires_grad` set to False). 
So I added a function to unfreeze the weights
```
    def on_stage_start(self, stage, epoch):
        """Gets called at the beginning of each epoch.

        Arguments
        ---------
        stage : sb.Stage
            One of sb.Stage.TRAIN, sb.Stage.VALID, or sb.Stage.TEST.
        epoch : int
            The currently-starting epoch. This is passed
            `None` during the test stage.
        """
        # Set up statistics trackers for this stage
        # In this case, we would like to keep track of the word error rate (wer)
        if stage == sb.Stage.TRAIN:
            for module in [self.modules.enc, self.modules.emb, self.modules.dec, self.modules.seq_lin]:
                for p in module.parameters():
                    p.requires_grad = True
```
I think this should be added to the fine-tune tutorial as well

**Comment by mravanelli** [comment]
Thank you very much for your feedback! Samuele, could you please take a
look into that?

**Comment by TParcollet** [comment]
@popcornell

**Comment by popcornell** [comment]
I am going to fix this today, thank you @ziz19  for pointing this out

**Comment by ziz19** [comment]
@popcornell thank you so much for taking care of this. I've seen that the `require_grad` has been added. I see that the data pipeline for fine-tuning is still using `dataset.add_dynamic_item(sb.dataio.dataio.read_audio, takes="file_path", provides="signal")` where it doesn't call any `AudioNormalizer` class. 
In `speechbrain/pretrained/interface.py`, the `load_audio` method actually calls the `AudioNormalizer`. So the audio input was normalized before passing into the ASR module. Do you think it might be necessary to change the data pipeline, or does it not matter?

**Comment by popcornell** [comment]
@ziz19 thank you for pointing this out. 

> In the fine-tuning section, the audio is loaded using dataset.add_dynamic_item(sb.dataio.dataio.read_audio, takes="file_path", provides="signal"), and no normalizer is used within the function.
> 
> Since the decoding relies on normalized audio, wouldn't it be more natural to fine-tune the model based on normalized audio rather than raw audio?

Well the model is fine-tuned on MiniLibrispeech which is a sub-set of librispeech so in this instance it does not make any difference. IMO the user should be responsible to take care of any mismatch knowing where the original model has been trained on. The Audio Normalizer is more for inference IMO. When fine-tuning you want to be in complete control of the data pipeline and if you add the Audio Normalizer you are not anymore (unless you know exactly what it does). 

BTW I ve added you in an Acknoledgment Section at the end of the tutorial. I can add your full name if you want.

**Comment by ziz19** [comment]
@popcornell Thank you for much for the explanation. Github username is perfectly fine! Speechbrain is so much easier to use compared to other frameworks. Thank you all for the hard work! I'm keep using Speechbrain in the future.
